ODIN_II/SRC/adders.cpp

```diff
@@ -1121,27 +1122,25 @@ void free_op_nodes(nnode_t* node) {
  * (function: match_pins)
  *-------------------------------------------------------------------------*/
 int match_pins(nnode_t* node, nnode_t* next_node) {
-    int flag = 0;
-    int i, j;
     long id;
-    for (i = 0; i < node->num_input_pins; i++) {
-        flag = 0;
-        id = node->input_pins[i]->net->driver_pin->unique_id;
-        for (j = 0; j < next_node->num_input_pins; j++) {
-            if (id == next_node->input_pins[j]->net->driver_pin->unique_id) {
-                flag = 1;
-                break;
-            }
-            if (j == next_node->num_input_pins - 1) {
-                flag = -1;
-                break;
+    for (int i = 0; i < node->num_input_pins; i++) {
+        for (int j = 0; j < node->input_pins[i]->net->num_driver_pins; j++) {
+            bool found = false;
+            id = node->input_pins[i]->net->driver_pins[i]->unique_id;
```

Comment:
can you declare id here rather than outside the loop (while you are changing it)

Reply:
Happily